http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -771,6 +774,7 @@ private JettyResponse.CompletionCallback onCompletion(Span span, Request request
                 recordRequestComplete(stats, request, requestSize.get(), requestStart, response, responseStart);
             }
             span.setAttribute(HttpIncubatingAttributes.HTTP_RESPONSE_BODY_SIZE, response.getBytesRead());
+            span.end();
```

Comment:
The `finally` in `execute()` above needs to be handled:

```
finally {
            span.end();
        }
```

Reply:
not really, the span is completed by callback

Reply:
```
public <T, E extends Exception> T execute(Request request, ResponseHandler<T, E> responseHandler)
            throws E
    {
        request = applyRequestFilters(request);
        Span span = startSpan(request);
        request = injectTracing(request, span);
        try {
            return doExecute(request, responseHandler, span);
        }
        catch (Throwable t) {
            span.setStatus(StatusCode.ERROR, t.getMessage());
            span.recordException(t, Attributes.of(ExceptionAttributes.EXCEPTION_ESCAPED, true));
            throw t;
        }
        finally {
            span.end();
        }
```